### PR TITLE
Add missing next method to iterators returned by Pool.imap

### LINKED ISSUE #1924: next method missing from iterators returned by multiprocessing.Pool.imap in Python 3 stubs
(The following is the exactly same as #1923. I created this separate issue because I found two different problems through the following code.)

```python
import time
import multiprocessing


def func(x):
    # type: (int) -> int
    time.sleep(x)
    return x


def main():
    # type: () -> None
    pool = multiprocessing.Pool(2)
    ite = pool.imap(func, range(10), chunksize=1)

    try:
        while True:
            x = ite.next(timeout=3)
            print(x)
    except multiprocessing.TimeoutError:
        print('timeout')
    

if __name__ == '__main__':
    main()
```

This code correctly works as follows.

```
$ python --version
Python 3.6.4
$ python foo.py
0
1
2
3
4
5
6
timeout
```

However, `mypy` (GitHub master) fails as follows:

```
$ mypy --version
mypy 0.570-dev-c594761aa1dbe76a3a0bc199b03fbf684ce588d8
$ mypy --python-version 3.6 foo.py
foo.py:18: error: "Iterable[Any]" has no attribute "next"
foo.py:20: error: Module has no attribute "TimeoutError"
```

FYI: when we specify `--python-version 2.7`, `mypy` says nothing against the code.

### PR DESCRIPTION
**Pull request by iwiwi**
Fixes #1924.

It adds `next(timeout)` method (https://docs.python.org/3.6/library/multiprocessing.html#multiprocessing.pool.Pool.imap) to iterators returned by `imap` and `imap_unordered` methods of `multiprocessing.Pool` and `multiprocessing.pool.ThreadPool`.

### CODE REVIEW

File: stdlib/3/multiprocessing/pool.pyi [hunk 1]
```diff
@@ -43,11 +46,11 @@ class Pool(ContextManager[Pool]):
     def imap(self,
              func: Callable[..., Any],
              iterable: Iterable[Any] = ...,
-             chunksize: Optional[int] = ...) -> Iterable[Any]: ...
+             chunksize: Optional[int] = ...) -> IMapIterator: ...
     def imap_unordered(self,
                        func: Callable[..., Any],
                        iterable: Iterable[Any] = ...,
-                       chunksize: Optional[int] = ...) -> Iterable[Any]: ...
+                       chunksize: Optional[int] = ...) -> IMapIterator: ...
```
Comment (JelleZijlstra) [inline stdlib/3/multiprocessing/pool.pyi]: This technically returns an IMapUnorderedIterator (a subclass of IMapIterator), but there doesn't seem to be any interface difference with IMapIterator, so I think we can safely ignore that.

File: stdlib/3/multiprocessing/pool.pyi [hunk 2]
```diff
@@ -15,6 +15,9 @@ class AsyncResult():
     def ready(self) -> bool: ...
     def successful(self) -> bool: ...
 
+class IMapIterator(Iterable[Any]):
+    def next(self, timeout: float = ...) -> Any: ...
```
Comment (JelleZijlstra) [inline stdlib/3/multiprocessing/pool.pyi]: Should be `Optional[float]`.

Comment (iwiwi) [reply stdlib/3/multiprocessing/pool.pyi]: Thanks for pointing it out!

File: stdlib/3/multiprocessing/pool.pyi (hunk 1 above)
Comment (iwiwi) [reply stdlib/3/multiprocessing/pool.pyi]: Thanks for checking, my opinion is the same.

**Comment by iwiwi** [comment]
ping? @JelleZijlstra

**Comment by iwiwi** [comment]
Thank you for your quick review on my three PRs!

**Pull request merged**